pilot/pkg/networking/core/v1alpha3/route/route.go

```diff
@@ -512,6 +512,13 @@ func applyHTTPRouteDestination(
 				},
 				Substitution: fullURI,
 			}
+		} else if regexRewrite := in.Rewrite.GetUriRegexRewrite(); regexRewrite != nil {
```

Comment:
can we remove the `FULLREPLACE` and generate it with regex directly in 
```
pilot/pkg/config/kube/gateway/conversion.go
1212:                   rewrite.Uri = fmt.Sprintf("%%FULLREPLACE()%%%s", *filter.Path.ReplaceFullPath)
```

Reply:
Both, change conversion.go to use native regex support instead of the ReplaceFullPath hack, then remove L508 since its now unused

Reply:
opened a new PR for this one as well https://github.com/istio/istio/pull/45423